### ISSUE TITLE
Fix false positives for Less variables in length-zero-no-unit

### ISSUE THREAD

**Issue by cordis-dev** [stylelint/stylelint#4200]
> Clearly describe the bug

Automatic fix (` --fix`) for `length-zero-no-unit` does nothing when working with Less variables.

> Which rule, if any, is the bug related to?

`length-zero-no-unit`

> What CSS is needed to reproduce the bug?

```less
@variable: 0px;
```

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "length-zero-no-unit": true
  }
}
```
> Which version of stylelint are you using?

`10.1.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

Node.js

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Yes, it's related to Less variables.

> What did you expect to happen?

Input file to be fixed:
```less
@variable: 0;
```

> What actually happened (e.g. what warnings or errors did you get)?

Input file did not change.

**Comment by evilebottnawi** [comment]
It is not safe, because we don't know how you will use variable, for example `line-height: 0` and `line-height: 0px` is difference values and means difference things

**Comment by hudochenkov** [comment]
The rule reports violation, so it should fix as well. Marking this as “help wanted”.

**Comment by vankop** [comment]
@hudochenkov However @evilebottnawi  is right, this values will have different meaning. Don't we need to change rule behavior?

**Comment by hudochenkov** [comment]
I think we do need to change the rule and don't check non-standard properties.

**Comment by evilebottnawi** [comment]
I think we should mark this as bug :smile:

**Comment by jeddy3** [comment]
It looks like we [already have a heuristic for Less variables](https://github.com/stylelint/stylelint/blob/master/lib/utils/isStandardSyntaxAtRule.js#L23), so it should just be the case of making use of this util in the rule.
